omod/src/main/java/org/openmrs/module/queue/web/QueueEntryTransitionRestController.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.web;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+
+import org.openmrs.Concept;
+import org.openmrs.api.APIException;
+import org.openmrs.module.queue.api.QueueServicesWrapper;
+import org.openmrs.module.queue.model.Queue;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.openmrs.module.queue.model.QueueEntryTransition;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * The main controller that exposes additional end points for order entry
+ */
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/queue-entry-transition")
+public class QueueEntryTransitionRestController extends BaseRestController {
+	
+	public static final String QUEUE_ENTRY_TO_TRANSITION = "queueEntryToTransition";
+	
+	public static final String TRANSITION_DATE = "transitionDate";
+	
+	public static final String NEW_QUEUE = "newQueue";
+	
+	public static final String NEW_STATUS = "newStatus";
+	
+	public static final String NEW_PRIORITY = "newPriority";
+	
+	private final QueueServicesWrapper services;
+	
+	@Autowired
+	public QueueEntryTransitionRestController(QueueServicesWrapper services) {
+		this.services = services;
+	}
+	
+	@RequestMapping(method = { RequestMethod.PUT, RequestMethod.POST })
+	@ResponseStatus(HttpStatus.OK)
+	public void transitionQueueEntry(@RequestBody Map<String, String> body) {
+		QueueEntryTransition transition = new QueueEntryTransition();
+		
+		// Queue Entry to Transition
+		String queueEntryUuid = body.get(QUEUE_ENTRY_TO_TRANSITION);
+		Optional<QueueEntry> queueEntryOptional = services.getQueueEntryService().getQueueEntryByUuid(queueEntryUuid);
+		if (!queueEntryOptional.isPresent()) {
+			throw new APIException(QUEUE_ENTRY_TO_TRANSITION + " is a required parameter");
+		}
+		transition.setQueueEntryToTransition(queueEntryOptional.get());
```

Comment:
I think the preferred unwrapping pattern is something like:

```java
QueueEntry queueEntry = queueEntryOptional.getOrThrow(() -> new APIException());
```

Reply:
Done.  I just find it uglier to read.  It would be happier if these optionals were not in the API.